controller/replica_controller.go

```diff
@@ -215,6 +230,19 @@ func (rc *ReplicaController) syncReplica(key string) (err error) {
 	}
 
 	if replica.DeletionTimestamp != nil {
+		if orphaned, err := rc.isOrphaned(replica); err != nil {
+			return err
+		} else if orphaned {
+			// adopt replica
+			replica.Spec.DataPath = ""
+			replica.Spec.NodeID = rc.controllerID
+			replica.Spec.OwnerID = rc.controllerID
+			_, err = rc.ds.UpdateReplica(replica)
+			if err == nil {
+				rc.eventRecorder.Eventf(replica, v1.EventTypeNormal, EventReasonOrphaned, "Adopted replica %v", replica.Name)
```

Comment:
This message is not helpful to the end user. They don't care who adopts the replica (and there is no information here to show who adopted it anyway), but the implication of this adoption is the replica content won't be cleaned up from the original node. That's something the user needs to know, in case e.g. they want to repurpose the node or add it back.